js/packages/components/chat/message/Reactions.tsx

```diff
@@ -0,0 +1,287 @@
+	useConversationMembersDict,
+	useMessengerClient,
+	useThemeColor,
+} from '@berty/hooks'
+import { selectInteraction } from '@berty/redux/reducers/messenger.reducer'
+import { getEmojiByName } from '@berty/utils/emojis/emojis'
+
+import { UnifiedText } from '../../shared-components/UnifiedText'
+
+const MemberItem: React.FC<{ member: any; divider: boolean }> = ({ member, divider = true }) => {
+	const { row, margin, padding, border } = useStyles()
+	const colors = useThemeColor()
+
+	return (
+		<View>
+			<View style={[row.left, row.item.justify, padding.vertical.small, { flexShrink: 1 }]}>
+				<ContactAvatar size={50} publicKey={member?.publicKey} />
+				<UnifiedText
+					numberOfLines={1}
+					style={[margin.left.small, row.item.justify, { flexShrink: 1 }]}
+				>
+					{member?.displayName!}
+				</UnifiedText>
+			</View>
+			{divider && (
+				<View style={[border.scale(0.6), { borderColor: `${colors['secondary-text']}90` }]} />
+			)}
+		</View>
+	)
+}
+
+const AnimatedNumber: React.FC<{ count: number; style: StyleProp<TextStyle> }> = ({
+	count,
+	style,
+}) => (
+	<View>
+		<UnifiedText style={style}>{count}</UnifiedText>
+	</View>
+)
+
+const ReactionList: FC<{
+	reactions: berty.messenger.v1.Interaction.IReactionView[]
+	emoji: string | null | undefined
+	cid: string
+	convPk: string
+	close: () => void
+}> = ({ reactions, emoji, cid, convPk }) => {
+	const colors = useThemeColor()
+	const { padding, border, margin, text } = useStyles()
+	const [currentEmoji, setCurrentEmoji] = useState<string | null | undefined>(emoji)
+	const [dataSourceCords, setDataSourceCords] = useState<number[] | undefined>()
+	const [userList, setUserList] = useState<(berty.messenger.v1.IMember | undefined)[]>([])
+	const scrollViewRef = createRef<ScrollView>()
+	const client = useMessengerClient()
+	const members = useConversationMembersDict(convPk)
+
+	const getReactionList = useCallback(async () => {
+		const r = (await client?.interactionReactionsForEmoji({
+			emoji: currentEmoji,
+			interactionCid: cid,
+		})) as berty.messenger.v1.InteractionReactionsForEmoji.Reply
+
+		setUserList(
+			Object.values(members).filter(member =>
+				r?.reactions.find(reaction => reaction.memberPublicKey === member?.publicKey),
+			) || [],
+		)
+	}, [cid, client, currentEmoji, members])
+
+	useEffect(() => {
+		setCurrentEmoji(emoji)
+	}, [emoji])
+
+	useEffect(() => {
+		getReactionList().then()
+	}, [getReactionList, currentEmoji])
+
+	useEffect(() => {
+		const index = reactions.findIndex(({ emoji }) => emoji === currentEmoji)
+		setTimeout(
+			() => {
+				scrollViewRef.current?.scrollTo({
+					x: dataSourceCords ? dataSourceCords[index] : 0,
+					y: 0,
+					animated: true,
+				})
+			},
+			(dataSourceCords || []).length === reactions.length ? 0 : 50,
+		)
+		scrollViewRef.current?.scrollTo({
+			x: dataSourceCords ? dataSourceCords[index] : 0,
+			y: 0,
+			animated: true,
+		})
+	}, [currentEmoji, dataSourceCords, reactions, scrollViewRef])
```

Comment:
please add comment explaining why we need the `setTimeout`

Reply:
I don't know why i just get the last version of this file to re-enable the feature